### DISCUSSION TITLE
Error on make after update

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#25384] [Q&A General] [answered]
I performed a mamba update all and updates to moose last night. Upon attempting to perform make of the modules, I am getting the following error:

fatal error: DistributedForce.h: No such file or directory

What is the possible solutions to avoid this error?

**Comment by milljm** [comment]
Without knowing what mamba packages you updated to, you may have run into the following (just a guess): https://github.com/idaholab/moose/discussions/25358

**Comment by TLWise** [comment]
The potential solution as presented did not fix the issue. Here is a more detail to the error:
/home/twise/projects/moose/modules/navier_stokes/src/kernels/DistributedPower.C:10:10: fatal error: DistributedPower.h: No such file or directory
   10 | #include "DistributedPower.h"
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [/home/twise/projects/moose/framework/build.mk:144: /home/twise/projects/moose/modules/navier_stokes/build/unity_src/kernels_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....

**Reply by GiudGiud** [reply]
Hello

This looks odd. Does the file exist? Look in navier_stokes/include/kernels for DistributedPower.h (your current error) and DistributedForce.h (the first one you pasted)

Did the clone of the moose repo go well? What does `git diff` and `git status` return ?

**Comment by TLWise** [comment]
The DistributedForce.h  file does not exit, however there is a file DistributedForce.C. I'll check the git diff and git status as soon as I have  an opportunity. Thanks.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by GiudGiud** [reply]
I think the clone must have failed to download it. It s in the repo, I checked

**Reply by GiudGiud** [reply]
either that or a gremlin deleted it from your folder

**Comment by TLWise** [comment]
It appears during the update --all process most of the .h files in this directory were deleted: modules/navier_stokes/include/kernels/

**Reply by milljm** [reply]
If we're talking about `mamba update --all`, that process does not have the ability to delete files anywhere other than in $CONDA_PREFIX.

But also, `mamba update --all` is not the correct solution for this round of updates. Please visit the sticky post for more details on how to properly 'update' to the latest moose packages: https://github.com/idaholab/moose/discussions/25358

**Comment by TLWise** [comment]
when I run the "make" command, I get the following message: /home/twise/projects/moose/framework/app.mk:209: warning: ignoring old recipe for target '/home/twise/projects/moose/modules/navier_stokes/build/header_symlinks/DistributedForce.h'

**Reply by GiudGiud** [reply, marked as answer]
make clobberall will help for that

**Reply by TLWise** [reply]
I used the clobberall, that appears to have got MOOSE functional again. With all that I have attempted to try, all of my previous folders and work in the module folders is no longer there. Luckily I have some backups on a separate disk to retrieve some of my work.

**Reply by GiudGiud** [reply]
I'm sorry to hear about this inconvenience.
`make clobberall` is fairly safe with regards to deletions iirc. `git clean -xfd` is a lot more dangerous in that regard
